test/tls-session-reuse.js

```diff
@@ -130,3 +130,103 @@ test('TLS should reuse sessions', { skip: nodeMajor < 11 }, t => {
 
   t.end()
 })
+
+test('A pool should be able to reuse TLS sessions between clients', { skip: nodeMajor < 11 }, t => {
```

Comment:
What happens if this is used in node v10? Is it crashing? We should document and add a a test for it.

Reply:
i don't immediately see why this wouldn't run in node 10 but i could easily be missing something obvious

Reply:
@ronag @delvedor @mcollina do you remember why that was done for the [Client's session reuse](https://github.com/nodejs/undici/pull/358)?
This code covers the same path so I decided to reuse the node version skip.

Reply:
The [`session` event](https://nodejs.org/api/tls.html#tls_event_session) has been added in Node v11.

Reply:
Thank you!